jobrunner/lib/database.py

```diff
@@ -28,10 +28,9 @@ def insert(item):
     get_connection().execute(sql, encode_field_values(fields, item))
 
 
-def update(item, update_fields=None):
+def update(item):
     assert item.id
-    if update_fields is None:
-        update_fields = [f.name for f in dataclasses.fields(item)]
+    update_fields = [f.name for f in dataclasses.fields(item)]
```

Comment:
Would it be worth excluding 'cancelled' field from this generic update?

It seems to me that only the sync thread can write cancelled, and main thread should always read, that might be a nicer solution that relying on a reset on next sync?

Reply:
Good. That's much better, thank you.

I've rewritten the commit to incorporate that improvement to make it easier for you to review the change in its entirety. Nothing else has changed since you reviewed.

https://github.com/opensafely-core/job-runner/pull/279/commits/ba193b015cc47b6e62959bae5389fbf2fec93e14

Reply:
Looks good!